File: robin/python/level1/study6/hate_same_num.py
def solution(arr):
    answer = []
    for idx, val in enumerate(arr):

        if idx == len(arr) - 1:
            answer.append(arr[len(arr) - 1])
            break

        if val != arr[idx + 1]:
            answer.append(val)

    return answer


if __name__ == '__main__':
    arr = [4, 4, 4, 3, 3]
    print(solution(arr))

File: robin/java/src/level1/study6/HateSameNum.java
package level1.study6;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

public class HateSameNum {

    public static List<Integer> solution(int[] arr) {
        List<Integer> answer = new ArrayList<>();

        for (int i = 0; i < arr.length; i++) {

            if (i == (arr.length - 1)) {
                answer.add(arr[i]);
                break;
            }

            if (arr[i] != arr[i + 1]) {
                answer.add(arr[i]);
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        int[] arr = {1, 1, 3, 3, 0, 1, 1};
        System.out.println(solution(arr).toString());
    }
}
